ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -216,6 +255,102 @@ KEXT_STATIC void UseMainForkIfNamedStream(
     }
 }
 
+static void ResizePendingRenames(uint32_t newMaxPendingRenames)
+{
+    PendingRenameOperation* newArray = Memory_AllocArray<PendingRenameOperation>(newMaxPendingRenames);
```

Comment:
Does this memory get freed if newMaxPendingRenames < s_maxPendingRenames

Reply:
> Does this memory get freed if newMaxPendingRenames < s_maxPendingRenames

Looks like I messed that up, thanks for catching it! This code went through a number of iterations, and it looks like I lost sight of that edge case.

> What prevents newMaxPendingRenames from becoming arbitrary large?

Reply:
> > Does this memory get freed if newMaxPendingRenames < s_maxPendingRenames
> 
> Looks like I messed that up, thanks for catching it! This code went through a number of iterations, and it looks like I lost sight of that edge case.

This is now fixed.